In [1]:
# import packages

import numpy as np
import pandas as pd
import seaborn as sns

import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
import folium


from scipy import stats
from scipy.stats import norm
from sklearn.dummy import DummyRegressor
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler, OneHotEncoder, OrdinalEncoder
from statsmodels.formula.api import ols
from sklearn.metrics import mean_squared_error as MSE


In [2]:
# Import dataset
data = pd.read_csv("data/kc_house_data.csv")
data

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
0          7140     1.0         NO        NO  ...       PUBLIC       1180   
1          6703     1.0         NO        NO  ...       PUBLIC       1570   
2          6156     1.0         NO        NO  ...       PUBLIC       1580   
3          1400     2.0         NO        NO  ...       PUBLIC       1090   
4           758     2.0         NO        NO  ...       PUBLIC       1120   
...         ...     ...        ...       ...  ...          ...        ...   
30150      4000     1.5         NO        NO  ...       PUBLIC       1600   
30151      5800     2.0         NO        NO  ...       PUBLIC       2020   
30152      3600     1.0         NO        NO  ...       PUBLIC        940   
30153      2889     2.0         NO        NO  ...       PUBLIC       1830   
30154     11058     1.0         NO        NO  ...       PUBLIC       1200   

      sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
0                 0           0         40     1969             0   
1              1570           0        240     1950             0   
2              1580           0          0     1956             0   
3              1070         200        270     2010             0   
4               550         550         30     2012             0   
...             ...         ...        ...      ...           ...   
30150          1130           0        210     1921             0   
30151             0           0        520     2011             0   
30152           920         240        110     1995             0   
30153           740         480        100     2006             0   
30154             0         420          0     1965             0   

                                                 address        lat       long  
0      2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.19052  
1      11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.35591  
2      8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.22520  
3      4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.29020  
4      2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.07188  
...                                                  ...        ...        ...  
30150  4673 Eastern Avenue North, Seattle, Washington...  47.664740 -122.32940  
30151  4131 44th Avenue Southwest, Seattle, Washingto...  47.565610 -122.38851  
30152  910 Martin Luther King Jr Way, Seattle, Washin...  47.610395 -122.29585  
30153  17127 114th Avenue Southeast, Renton, Washingt...  47.449490 -122.18908  
30154  18615 7th Avenue South, Burien, Washington 981...  47.435840 -122.32634  

[30155 rows x 25 columns]

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30155 entries, 0 to 30154
Data columns (total 25 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             30155 non-null  int64  
 1   date           30155 non-null  object 
 2   price          30155 non-null  float64
 3   bedrooms       30155 non-null  int64  
 4   bathrooms      30155 non-null  float64
 5   sqft_living    30155 non-null  int64  
 6   sqft_lot       30155 non-null  int64  
 7   floors         30155 non-null  float64
 8   waterfront     30155 non-null  object 
 9   greenbelt      30155 non-null  object 
 10  nuisance       30155 non-null  object 
 11  view           30155 non-null  object 
 12  condition      30155 non-null  object 
 13  grade          30155 non-null  object 
 14  heat_source    30123 non-null  object 
 15  sewer_system   30141 non-null  object 
 16  sqft_above     30155 non-null  int64  
 17  sqft_basement  30155 non-null  int64  
 18  sqft_g

In [4]:
data.corr()

id     price  bedrooms  bathrooms  sqft_living  sqft_lot  \
id             1.000000 -0.034184 -0.006306  -0.012094    -0.027932 -0.119101   
price         -0.034184  1.000000  0.289204   0.480401     0.608521  0.085730   
bedrooms      -0.006306  0.289204  1.000000   0.589273     0.637874  0.003306   
bathrooms     -0.012094  0.480401  0.589273   1.000000     0.772677  0.035886   
sqft_living   -0.027932  0.608521  0.637874   0.772677     1.000000  0.119563   
sqft_lot      -0.119101  0.085730  0.003306   0.035886     0.119563  1.000000   
floors         0.032043  0.180576  0.147592   0.404412     0.304240 -0.032097   
sqft_above    -0.023216  0.538651  0.547164   0.674924     0.883984  0.129231   
sqft_basement -0.014662  0.245058  0.238502   0.260902     0.338460  0.004111   
sqft_garage   -0.007829  0.264169  0.319441   0.457022     0.511740  0.087169   
sqft_patio    -0.041625  0.313409  0.183439   0.327551     0.396030  0.155250   
yr_built       0.023071  0.096013  0.146191   0.443648     0.291694  0.001750   
yr_renovated  -0.029131  0.084786  0.014286   0.040631     0.038499  0.010049   
lat           -0.000691  0.063632  0.108758  -0.005225     0.102186  0.030020   
long           0.000479 -0.022509 -0.106689   0.017400    -0.087669 -0.034308   

                 floors  sqft_above  sqft_basement  sqft_garage  sqft_patio  \
id             0.032043   -0.023216      -0.014662    -0.007829   -0.041625   
price          0.180576    0.538651       0.245058     0.264169    0.313409   
bedrooms       0.147592    0.547164       0.238502     0.319441    0.183439   
bathrooms      0.404412    0.674924       0.260902     0.457022    0.327551   
sqft_living    0.304240    0.883984       0.338460     0.511740    0.396030   
sqft_lot      -0.032097    0.129231       0.004111     0.087169    0.155250   
floors         1.000000    0.448281      -0.248093     0.132656    0.125183   
sqft_above     0.448281    1.000000      -0.066801     0.560551    0.312117   
sqft_basement -0.248093   -0.066801       1.000000     0.026361    0.210500   
sqft_garage    0.132656    0.560551       0.026361     1.000000    0.216354   
sqft_patio     0.125183    0.312117       0.210500     0.216354    1.000000   
yr_built       0.544646    0.387448      -0.230226     0.447560    0.138409   
yr_renovated  -0.025449    0.010598       0.053634    -0.098603    0.056385   
lat           -0.218554    0.092258       0.059788     0.092346   -0.019724   
long           0.233781   -0.082744      -0.045208    -0.096861    0.025699   

               yr_built  yr_renovated       lat      long  
id             0.023071     -0.029131 -0.000691  0.000479  
price          0.096013      0.084786  0.063632 -0.022509  
bedrooms       0.146191      0.014286  0.108758 -0.106689  
bathrooms      0.443648      0.040631 -0.005225  0.017400  
sqft_living    0.291694      0.038499  0.102186 -0.087669  
sqft_lot       0.001750      0.010049  0.030020 -0.034308  
floors         0.544646     -0.025449 -0.218554  0.233781  
sqft_above     0.387448      0.010598  0.092258 -0.082744  
sqft_basement -0.230226      0.053634  0.059788 -0.045208  
sqft_garage    0.447560     -0.098603  0.092346 -0.096861  
sqft_patio     0.138409      0.056385 -0.019724  0.025699  
yr_built       1.000000     -0.239284 -0.207186  0.209878  
yr_renovated  -0.239284      1.000000  0.036956 -0.035631  
lat           -0.207186      0.036956  1.000000 -0.760766  
long           0.209878     -0.035631 -0.760766  1.000000

# Initial Look at Data to Get Information

In [5]:
data['sqft_above'].value_counts()

1200    282
1300    282
1060    271
1100    268
1250    265
       ... 
1799      1
1783      1
1767      1
7700      1
2049      1
Name: sqft_above, Length: 1187, dtype: int64

In [6]:
data['heat_source'].value_counts()

Gas                  20583
Electricity           6465
Oil                   2899
Gas/Solar               93
Electricity/Solar       59
Other                   20
Oil/Solar                4
Name: heat_source, dtype: int64

In [7]:
data['sewer_system'].value_counts()

PUBLIC                25777
PRIVATE                4355
PRIVATE RESTRICTED        6
PUBLIC RESTRICTED         3
Name: sewer_system, dtype: int64

In [8]:
data['sewer_system'].isnull()

0        False
1        False
2        False
3        False
4        False
         ...  
30150    False
30151    False
30152    False
30153    False
30154    False
Name: sewer_system, Length: 30155, dtype: bool

# One duplicate in ID, a few nulls in 'Sewer_system' and 'heat_source', could drop these since there are only a few

### Dropping the duplicated ID

In [9]:
#Check for duplicates
data[data.duplicated(subset=['id'])]

id       date      price  bedrooms  bathrooms  sqft_living  \
4846  1233100736  9/28/2021  2600000.0         3        4.0         3500   

      sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
4846      8455     2.0         NO        NO  ...       PUBLIC       2590   

     sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
4846          1410         500        290     2019             0   

                                                address       lat        long  
4846  12611 Northeast 87th Place, Kirkland, Washingt...  47.68069 -122.172135  

[1 rows x 25 columns]

In [10]:
#Verify that duplicate is actually a duplicate
data[data['id'].isin(['1233100736'])]

id       date      price  bedrooms  bathrooms  sqft_living  \
4845  1233100736  9/28/2021  2600000.0         3        4.0         3500   
4846  1233100736  9/28/2021  2600000.0         3        4.0         3500   

      sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
4845      8455     2.0         NO        NO  ...       PUBLIC       2590   
4846      8455     2.0         NO        NO  ...       PUBLIC       2590   

     sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
4845          1410         500        290     2019             0   
4846          1410         500        290     2019             0   

                                                address       lat        long  
4845  12611 Northeast 87th Place, Kirkland, Washingt...  47.68069 -122.172135  
4846  12611 Northeast 87th Place, Kirkland, Washingt...  47.68069 -122.172135  

[2 rows x 25 columns]

In [11]:
#Dropping duplicate row
data_clean = data.drop_duplicates(subset = ['id'])
data_clean

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
0          7140     1.0         NO        NO  ...       PUBLIC       1180   
1          6703     1.0         NO        NO  ...       PUBLIC       1570   
2          6156     1.0         NO        NO  ...       PUBLIC       1580   
3          1400     2.0         NO        NO  ...       PUBLIC       1090   
4           758     2.0         NO        NO  ...       PUBLIC       1120   
...         ...     ...        ...       ...  ...          ...        ...   
30150      4000     1.5         NO        NO  ...       PUBLIC       1600   
30151      5800     2.0         NO        NO  ...       PUBLIC       2020   
30152      3600     1.0         NO        NO  ...       PUBLIC        940   
30153      2889     2.0         NO        NO  ...       PUBLIC       1830   
30154     11058     1.0         NO        NO  ...       PUBLIC       1200   

      sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
0                 0           0         40     1969             0   
1              1570           0        240     1950             0   
2              1580           0          0     1956             0   
3              1070         200        270     2010             0   
4               550         550         30     2012             0   
...             ...         ...        ...      ...           ...   
30150          1130           0        210     1921             0   
30151             0           0        520     2011             0   
30152           920         240        110     1995             0   
30153           740         480        100     2006             0   
30154             0         420          0     1965             0   

                                                 address        lat       long  
0      2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.19052  
1      11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.35591  
2      8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.22520  
3      4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.29020  
4      2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.07188  
...                                                  ...        ...        ...  
30150  4673 Eastern Avenue North, Seattle, Washington...  47.664740 -122.32940  
30151  4131 44th Avenue Southwest, Seattle, Washingto...  47.565610 -122.38851  
30152  910 Martin Luther King Jr Way, Seattle, Washin...  47.610395 -122.29585  
30153  17127 114th Avenue Southeast, Renton, Washingt...  47.449490 -122.18908  
30154  18615 7th Avenue South, Burien, Washington 981...  47.435840 -122.32634  

[30154 rows x 25 columns]

### Removing the nulls from 'sewer_System' and 'heat_source'

In [12]:
#Drop columns with missing data
data_clean = data_clean.dropna(subset=['sewer_system', 'heat_source'])
data_clean

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sewer_system sqft_above  \
0          7140     1.0         NO        NO  ...       PUBLIC       1180   
1          6703     1.0         NO        NO  ...       PUBLIC       1570   
2          6156     1.0         NO        NO  ...       PUBLIC       1580   
3          1400     2.0         NO        NO  ...       PUBLIC       1090   
4           758     2.0         NO        NO  ...       PUBLIC       1120   
...         ...     ...        ...       ...  ...          ...        ...   
30150      4000     1.5         NO        NO  ...       PUBLIC       1600   
30151      5800     2.0         NO        NO  ...       PUBLIC       2020   
30152      3600     1.0         NO        NO  ...       PUBLIC        940   
30153      2889     2.0         NO        NO  ...       PUBLIC       1830   
30154     11058     1.0         NO        NO  ...       PUBLIC       1200   

      sqft_basement sqft_garage sqft_patio yr_built  yr_renovated  \
0                 0           0         40     1969             0   
1              1570           0        240     1950             0   
2              1580           0          0     1956             0   
3              1070         200        270     2010             0   
4               550         550         30     2012             0   
...             ...         ...        ...      ...           ...   
30150          1130           0        210     1921             0   
30151             0           0        520     2011             0   
30152           920         240        110     1995             0   
30153           740         480        100     2006             0   
30154             0         420          0     1965             0   

                                                 address        lat       long  
0      2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.19052  
1      11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.35591  
2      8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.22520  
3      4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.29020  
4      2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.07188  
...                                                  ...        ...        ...  
30150  4673 Eastern Avenue North, Seattle, Washington...  47.664740 -122.32940  
30151  4131 44th Avenue Southwest, Seattle, Washingto...  47.565610 -122.38851  
30152  910 Martin Luther King Jr Way, Seattle, Washin...  47.610395 -122.29585  
30153  17127 114th Avenue Southeast, Renton, Washingt...  47.449490 -122.18908  
30154  18615 7th Avenue South, Burien, Washington 981...  47.435840 -122.32634  

[30110 rows x 25 columns]

We have elected to drop the rows with missing data instead of imputing it because there are only 3 rows with missing values. 

# Separating Out Zipcode from the Address

In [13]:
data_clean['address']

0        2102 Southeast 21st Court, Renton, Washington ...
1        11231 Greenwood Avenue North, Seattle, Washing...
2        8504 South 113th Street, Seattle, Washington 9...
3        4079 Letitia Avenue South, Seattle, Washington...
4        2193 Northwest Talus Drive, Issaquah, Washingt...
                               ...                        
30150    4673 Eastern Avenue North, Seattle, Washington...
30151    4131 44th Avenue Southwest, Seattle, Washingto...
30152    910 Martin Luther King Jr Way, Seattle, Washin...
30153    17127 114th Avenue Southeast, Renton, Washingt...
30154    18615 7th Avenue South, Burien, Washington 981...
Name: address, Length: 30110, dtype: object

In [14]:
# This creates a new column with our zipcodes of the houses
data_clean['postal_code'] = data_clean['address'].str[-20:-15]
data_clean

<ipython-input-14-ad923d199017>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_clean['postal_code'] = data_clean['address'].str[-20:-15]


id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sqft_above sqft_basement  \
0          7140     1.0         NO        NO  ...       1180             0   
1          6703     1.0         NO        NO  ...       1570          1570   
2          6156     1.0         NO        NO  ...       1580          1580   
3          1400     2.0         NO        NO  ...       1090          1070   
4           758     2.0         NO        NO  ...       1120           550   
...         ...     ...        ...       ...  ...        ...           ...   
30150      4000     1.5         NO        NO  ...       1600          1130   
30151      5800     2.0         NO        NO  ...       2020             0   
30152      3600     1.0         NO        NO  ...        940           920   
30153      2889     2.0         NO        NO  ...       1830           740   
30154     11058     1.0         NO        NO  ...       1200             0   

      sqft_garage sqft_patio yr_built yr_renovated  \
0               0         40     1969            0   
1               0        240     1950            0   
2               0          0     1956            0   
3             200        270     2010            0   
4             550         30     2012            0   
...           ...        ...      ...          ...   
30150           0        210     1921            0   
30151           0        520     2011            0   
30152         240        110     1995            0   
30153         480        100     2006            0   
30154         420          0     1965            0   

                                                 address        lat  \
0      2102 Southeast 21st Court, Renton, Washington ...  47.461975   
1      11231 Greenwood Avenue North, Seattle, Washing...  47.711525   
2      8504 South 113th Street, Seattle, Washington 9...  47.502045   
3      4079 Letitia Avenue South, Seattle, Washington...  47.566110   
4      2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470   
...                                                  ...        ...   
30150  4673 Eastern Avenue North, Seattle, Washington...  47.664740   
30151  4131 44th Avenue Southwest, Seattle, Washingto...  47.565610   
30152  910 Martin Luther King Jr Way, Seattle, Washin...  47.610395   
30153  17127 114th Avenue Southeast, Renton, Washingt...  47.449490   
30154  18615 7th Avenue South, Burien, Washington 981...  47.435840   

            long  postal_code  
0     -122.19052        98055  
1     -122.35591        98133  
2     -122.22520        98178  
3     -122.29020        98118  
4     -122.07188        98027  
...          ...          ...  
30150 -122.32940        98103  
30151 -122.38851        98116  
30152 -122.29585        98122  
30153 -122.18908        98055  
30154 -122.32634        98148  

[30110 rows x 26 columns]

In [15]:
#verify no missing values -- sanity check 
data_clean.isna().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
greenbelt        0
nuisance         0
view             0
condition        0
grade            0
heat_source      0
sewer_system     0
sqft_above       0
sqft_basement    0
sqft_garage      0
sqft_patio       0
yr_built         0
yr_renovated     0
address          0
lat              0
long             0
postal_code      0
dtype: int64

### Any homes not within King County?

In [16]:
data_clean['postal_code'].value_counts(normalize=True)

98042    0.032946
98038    0.028462
98103    0.025274
98115    0.025241
98117    0.024842
           ...   
59501    0.000033
21702    0.000033
84104    0.000033
17922    0.000033
53214    0.000033
Name: postal_code, Length: 399, dtype: float64

It appears that addresses within King's County begin with Postal Code '98'. Are there any that don't being with '98'? Let's find out. 

In [17]:
#find any postal codes not within King County
prefixes = ['98']

data_clean[~data_clean.postal_code.str.startswith(tuple(prefixes))]

id        date      price  bedrooms  bathrooms  sqft_living  \
12     1797501124   6/25/2021   750000.0         3        2.0         1280   
53     7548300606    5/3/2022   960000.0         3        2.0         1280   
62     1934800106   8/24/2021   740000.0         2        2.0         1120   
159     856000595    7/8/2021  3730000.0         4        4.5         4820   
172    1336300219    2/9/2022   759900.0         2        2.0          960   
...           ...         ...        ...       ...        ...          ...   
30029  1978200468  10/28/2021  1480000.0         3        2.0         2050   
30044  9834201391   2/17/2022   520000.0         2        2.0          790   
30116  2768100152    1/1/2022   710000.0         1        2.0         1180   
30129  8584800130  11/18/2021   940000.0         2        2.0         1550   
30144  2267000442   12/1/2021   729950.0         2        2.0         1290   

       sqft_lot  floors waterfront greenbelt  ... sqft_above sqft_basement  \
12          964     3.0         NO        NO  ...       1280             0   
53         1221     2.0         NO        NO  ...        980           300   
62          734     3.0         NO        NO  ...       1120             0   
159       10800     2.0         NO        NO  ...       4820             0   
172         591     3.0         NO        NO  ...        960             0   
...         ...     ...        ...       ...  ...        ...           ...   
30029      1090     3.5         NO        NO  ...       1920           490   
30044       597     2.0         NO        NO  ...        520           270   
30116       616     3.0         NO        NO  ...        840           340   
30129      1026     2.5         NO        NO  ...       1310           470   
30144       720     3.0         NO        NO  ...       1180           360   

      sqft_garage sqft_patio yr_built yr_renovated  \
12              0        380     2021            0   
53              0        400     2019            0   
62              0        260     2021            0   
159           850          0     2006            0   
172             0        240     2021            0   
...           ...        ...      ...          ...   
30029         360        240     2021            0   
30044           0        210     2021            0   
30116           0        330     2021            0   
30129         230          0     2009            0   
30144         250        360     2021            0   

                                                 address        lat  \
12     5712 A Street, Omaha, Nebraska 68106, United S...  41.225811   
53     1820 South State Street, Vineland, New Jersey ...  39.460807   
62     1804 Spruce Street, McLeansboro, Illinois 6285...  38.075870   
159    626 13th Street, Union City, New Jersey 07087,...  40.761325   
172    11th Avenue, West Babylon, New York 11704, Uni...  40.719455   
...                                                  ...        ...   
30029  Avenue Cuts, 34 Ridge Rd, North Arlington, New...  40.782453   
30044  36th Avenue, Kenosha, Wisconsin 53142, United ...  42.548430   
30116  57th Street Lane NW, Rochester, Minnesota 5590...  44.081760   
30129  214 B, Mount Laurel, New Jersey 08054, United ...  39.955820   
30144  2954 Northwest 85th Street, Miami, Florida 331...  25.851697   

            long  postal_code  
12    -96.001397        68106  
53    -75.015361        08360  
62    -88.535760        62859  
159   -74.039020        07087  
172   -73.371935        11704  
...          ...          ...  
30029 -74.137642        07031  
30044 -87.851900        53142  
30116 -92.524610        55901  
30129 -74.881130        08054  
30144 -80.237790        33147  

[911 rows x 26 columns]

There are 911 rows that are not within the King's County area whose postal code does not being with '98'. Let's drop these homes from our dataset and focus on those solely within the county.

In [18]:
#find the indexes of the homes not within King County
prefixes = ['98']
non_king_postals = data_clean[~data_clean.postal_code.str.startswith(tuple(prefixes))].index
non_king_postals

#drop the rows whose postal code does not being with '98'
only_king = data_clean.drop(non_king_postals)
only_king.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7399300360   5/24/2022  675000.0         4        1.0         1180   
1  8910500230  12/13/2021  920000.0         5        2.5         2770   
2  1180000275   9/29/2021  311000.0         6        2.0         2880   
3  1604601802  12/14/2021  775000.0         3        3.0         2160   
4  8562780790   8/24/2021  592500.0         2        2.0         1120   

   sqft_lot  floors waterfront greenbelt  ... sqft_above sqft_basement  \
0      7140     1.0         NO        NO  ...       1180             0   
1      6703     1.0         NO        NO  ...       1570          1570   
2      6156     1.0         NO        NO  ...       1580          1580   
3      1400     2.0         NO        NO  ...       1090          1070   
4       758     2.0         NO        NO  ...       1120           550   

  sqft_garage sqft_patio yr_built yr_renovated  \
0           0         40     1969            0   
1           0        240     1950            0   
2           0          0     1956            0   
3         200        270     2010            0   
4         550         30     2012            0   

                                             address        lat       long  \
0  2102 Southeast 21st Court, Renton, Washington ...  47.461975 -122.19052   
1  11231 Greenwood Avenue North, Seattle, Washing...  47.711525 -122.35591   
2  8504 South 113th Street, Seattle, Washington 9...  47.502045 -122.22520   
3  4079 Letitia Avenue South, Seattle, Washington...  47.566110 -122.29020   
4  2193 Northwest Talus Drive, Issaquah, Washingt...  47.532470 -122.07188   

   postal_code  
0        98055  
1        98133  
2        98178  
3        98118  
4        98027  

[5 rows x 26 columns]

There might still be some zipcodes present within the dataset that are not within King's County, but are in a neighboring county whose postal code also begins with '98', but let's hone our data some more. 

## Creating new columns from existing data

### Numeric grade value column

In [19]:
only_king['grade'].unique()

array(['7 Average', '9 Better', '8 Good', '6 Low Average', '10 Very Good',
       '5 Fair', '11 Excellent', '12 Luxury', '4 Low', '13 Mansion',
       '3 Poor', '2 Substandard'], dtype=object)

In [20]:
#To get numeric grade code value
only_king['grade_code'] = only_king['grade'].str[0:2]

#To strip whitespace from the grade code
only_king['grade_code'] = only_king['grade_code'].str.strip()

In [21]:
#sanity check to make sure no white space
only_king['grade_code'].unique()

array(['7', '9', '8', '6', '10', '5', '11', '12', '4', '13', '3', '2'],
      dtype=object)

### Numeric condition value column

In [22]:
#To get numeric condition code
only_king['condition_code'] = only_king['condition']

# Replace the word to numeric value
only_king.condition_code.replace({'Poor': 1, 'Fair': 2, 'Average': 3, 'Good': 4,'Very Good': 5}, inplace=True)

### Converting Data Types of New Columns

In [23]:
#change postal code to numeric datatype
only_king['postal_code'] = only_king['postal_code'].astype(int)

#To convert to float
only_king['condition_code'] = only_king['condition_code'].astype(int)

#To convert to float
only_king['grade_code'] = only_king['grade_code'].astype(int)

## Feature engineering, want to know the price per sqft of livable sqft

In [24]:
only_king['price_sqft'] = only_king['price'] / only_king['sqft_living']
only_king

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
0          7140     1.0         NO        NO  ...         40     1969   
1          6703     1.0         NO        NO  ...        240     1950   
2          6156     1.0         NO        NO  ...          0     1956   
3          1400     2.0         NO        NO  ...        270     2010   
4           758     2.0         NO        NO  ...         30     2012   
...         ...     ...        ...       ...  ...        ...      ...   
30150      4000     1.5         NO        NO  ...        210     1921   
30151      5800     2.0         NO        NO  ...        520     2011   
30152      3600     1.0         NO        NO  ...        110     1995   
30153      2889     2.0         NO        NO  ...        100     2006   
30154     11058     1.0         NO        NO  ...          0     1965   

      yr_renovated                                            address  \
0                0  2102 Southeast 21st Court, Renton, Washington ...   
1                0  11231 Greenwood Avenue North, Seattle, Washing...   
2                0  8504 South 113th Street, Seattle, Washington 9...   
3                0  4079 Letitia Avenue South, Seattle, Washington...   
4                0  2193 Northwest Talus Drive, Issaquah, Washingt...   
...            ...                                                ...   
30150            0  4673 Eastern Avenue North, Seattle, Washington...   
30151            0  4131 44th Avenue Southwest, Seattle, Washingto...   
30152            0  910 Martin Luther King Jr Way, Seattle, Washin...   
30153            0  17127 114th Avenue Southeast, Renton, Washingt...   
30154            0  18615 7th Avenue South, Burien, Washington 981...   

             lat       long  postal_code  grade_code  condition_code  \
0      47.461975 -122.19052        98055           7               4   
1      47.711525 -122.35591        98133           7               3   
2      47.502045 -122.22520        98178           7               3   
3      47.566110 -122.29020        98118           9               3   
4      47.532470 -122.07188        98027           7               3   
...          ...        ...          ...         ...             ...   
30150  47.664740 -122.32940        98103           8               4   
30151  47.565610 -122.38851        98116           7               3   
30152  47.610395 -122.29585        98122           7               3   
30153  47.449490 -122.18908        98055           8               3   
30154  47.435840 -122.32634        98148           7               3   

       price_sqft  
0      572.033898  
1      332.129964  
2      107.986111  
3      358.796296  
4      529.017857  
...           ...  
30150  814.136126  
30151  650.000000  
30152  493.827160  
30153  301.556420  
30154  416.666667  

[29199 rows x 29 columns]

In [25]:
only_king['price_sqft'].describe()

count     29199.000000
mean        558.055358
std        3541.243279
min           6.920415
25%         357.754755
50%         487.654321
75%         641.876179
max      601000.000000
Name: price_sqft, dtype: float64

## Cleaning price to rid of outliers
- We utilize the Interqurtile range to rid outliers via the formula
- Below: Q1 - 1.5 IQR
- Above: Q3 + 1.5 IQR
- IQR = Q3 – Q1

In [26]:
only_king['price'].describe()

count    2.919900e+04
mean     1.112959e+06
std      8.954250e+05
min      2.736000e+04
25%      6.450000e+05
50%      8.680000e+05
75%      1.310000e+06
max      3.075000e+07
Name: price, dtype: float64

In [27]:
#Calculate Q1, Q2, Q3 and IQR
Q1 = np.percentile(only_king['price'], 25, interpolation = 'midpoint') 
Q2 = np.percentile(only_king['price'], 50, interpolation = 'midpoint') 
Q3 = np.percentile(only_king['price'], 75, interpolation = 'midpoint') 

print('Q1 25 percentile of the given data is, ', Q1)
print('Q2 50 percentile of the given data is, ', Q2)
print('Q3 75 percentile of the given data is, ', Q3)
  
IQR = Q3 - Q1 
print('Interquartile range (IQR) is', IQR)

Q1 25 percentile of the given data is,  645000.0
Q2 50 percentile of the given data is,  868000.0
Q3 75 percentile of the given data is,  1310000.0
Interquartile range (IQR) is 665000.0


In [28]:
#Find the lower and upper limits
lower_limit = Q1 - 1.5 * IQR
upper_limit = Q3 + 1.5 * IQR
print('Lower limit is', lower_limit)
print('Upper limit is', upper_limit)

Lower limit is -352500.0
Upper limit is 2307500.0


In [29]:
#Locate the homes greater than the upper limit
only_king[only_king['price'] >= 2307500]

id        date      price  bedrooms  bathrooms  sqft_living  \
27     5424700190   2/26/2022  4500000.0         4        3.0         2760   
36     1925059107   6/29/2021  2450000.0         4        3.5         2300   
43     1726059053   3/22/2022  3850000.0         5        3.5         4180   
84     6865200085    3/7/2022  2500000.0         4        3.5         3120   
118    5425700191  12/15/2021  3000000.0         3        1.5         2040   
...           ...         ...        ...       ...        ...          ...   
30100  3582050010   4/21/2022  2588000.0         5        4.5         3580   
30106  7588700215   4/11/2022  2875000.0         3        2.0         1900   
30126  1087500050  11/30/2021  3754500.0         4        5.5         5200   
30130  5456000475    3/9/2022  2435000.0         5        3.0         3920   
30140  8081900170   3/29/2022  2650000.0         4        3.5         3270   

       sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
27        13150     1.5         NO        NO  ...        160     1954   
36         8370     2.0         NO        NO  ...        110     1950   
43       209959     1.0         NO        NO  ...          0     2016   
84         3801     2.0         NO        NO  ...        380     2021   
118       14284     1.0         NO        NO  ...          0     1952   
...         ...     ...        ...       ...  ...        ...      ...   
30100      5719     2.0         NO        NO  ...        220     2019   
30106      8800     1.0         NO        NO  ...         80     1953   
30126     10790     2.0         NO        NO  ...        400     2021   
30130      8414     1.0         NO        NO  ...        350     1954   
30140      9200     2.0         NO        NO  ...         40     1925   

      yr_renovated                                            address  \
27               0  8637 Northeast 6th Street, Medina, Washington ...   
36            1995  9411 Northeast 32nd Street, Clyde Hill, Washin...   
43               0  10919 Northeast 164th Lane, Bothell, Washingto...   
84               0  1305 North 50th Street, Seattle, Washington 98...   
118              0  2539 82nd Avenue Northeast, Medina, Washington...   
...            ...                                                ...   
30100            0  21802 Southeast 5th Place, Sammamish, Washingt...   
30106            0  8322 16th Avenue Northwest, Seattle, Washingto...   
30126            0  12237 Northeast 66th Street, Kirkland, Washing...   
30130            0  9823 Mercerwood Drive, Mercer Island, Washingt...   
30140            0  3303 Northwest 72nd Street, Seattle, Washingto...   

             lat        long  postal_code  grade_code  condition_code  \
27     47.615710 -122.223990        98039           9               4   
36     47.639005 -122.214107        98004           9               3   
43     47.745460 -122.194980        98011           8               3   
84     47.664785 -122.341635        98103           8               3   
118    47.633020 -122.232570        98039           7               4   
...          ...         ...          ...         ...             ...   
30100  47.604675 -122.048760        98074           9               3   
30106  47.689410 -122.377560        98117           8               3   
30126  47.664625 -122.175870        98033          10               3   
30130  47.573195 -122.207510        98040           9               4   
30140  47.680710 -122.400025        98117           9               4   

        price_sqft  
27     1630.434783  
36     1065.217391  
43      921.052632  
84      801.282051  
118    1470.588235  
...            ...  
30100   722.905028  
30106  1513.157895  
30126   722.019231  
30130   621.173469  
30140   810.397554  

[1874 rows x 29 columns]

There are 1874 rows of data that have housing pricing greater than $2,307,500.

In [30]:
#Create a copy dataset
df_clean = only_king.copy()

#Filter out the homes beyond the upper limit of price
df_clean = df_clean[df_clean['price'] <= 2307500]
df_clean

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
0          7140     1.0         NO        NO  ...         40     1969   
1          6703     1.0         NO        NO  ...        240     1950   
2          6156     1.0         NO        NO  ...          0     1956   
3          1400     2.0         NO        NO  ...        270     2010   
4           758     2.0         NO        NO  ...         30     2012   
...         ...     ...        ...       ...  ...        ...      ...   
30150      4000     1.5         NO        NO  ...        210     1921   
30151      5800     2.0         NO        NO  ...        520     2011   
30152      3600     1.0         NO        NO  ...        110     1995   
30153      2889     2.0         NO        NO  ...        100     2006   
30154     11058     1.0         NO        NO  ...          0     1965   

      yr_renovated                                            address  \
0                0  2102 Southeast 21st Court, Renton, Washington ...   
1                0  11231 Greenwood Avenue North, Seattle, Washing...   
2                0  8504 South 113th Street, Seattle, Washington 9...   
3                0  4079 Letitia Avenue South, Seattle, Washington...   
4                0  2193 Northwest Talus Drive, Issaquah, Washingt...   
...            ...                                                ...   
30150            0  4673 Eastern Avenue North, Seattle, Washington...   
30151            0  4131 44th Avenue Southwest, Seattle, Washingto...   
30152            0  910 Martin Luther King Jr Way, Seattle, Washin...   
30153            0  17127 114th Avenue Southeast, Renton, Washingt...   
30154            0  18615 7th Avenue South, Burien, Washington 981...   

             lat       long  postal_code  grade_code  condition_code  \
0      47.461975 -122.19052        98055           7               4   
1      47.711525 -122.35591        98133           7               3   
2      47.502045 -122.22520        98178           7               3   
3      47.566110 -122.29020        98118           9               3   
4      47.532470 -122.07188        98027           7               3   
...          ...        ...          ...         ...             ...   
30150  47.664740 -122.32940        98103           8               4   
30151  47.565610 -122.38851        98116           7               3   
30152  47.610395 -122.29585        98122           7               3   
30153  47.449490 -122.18908        98055           8               3   
30154  47.435840 -122.32634        98148           7               3   

       price_sqft  
0      572.033898  
1      332.129964  
2      107.986111  
3      358.796296  
4      529.017857  
...           ...  
30150  814.136126  
30151  650.000000  
30152  493.827160  
30153  301.556420  
30154  416.666667  

[27325 rows x 29 columns]

In [31]:
df_clean['price'].median()

835000.0

In [32]:
#Wanted to see where price/sqft was equal to 601k
df_clean.loc[df_clean['price_sqft'] == 601000]

id        date      price  bedrooms  bathrooms  sqft_living  \
14977  1549500215  12/17/2021  1803000.0         4        4.0            3   

       sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
14977    326701     2.0         NO        NO  ...          0     2021   

      yr_renovated                                            address  \
14977            0  31072 Northeast 165th Street, Duvall, Washingt...   

            lat       long  postal_code  grade_code  condition_code  \
14977  47.74529 -121.92362        98019           9               3   

       price_sqft  
14977    601000.0  

[1 rows x 29 columns]

### Since this is for a family, we want at least 2 bedrooms and at least 1 bathroom in the house, only one foor and houses built after 1977 due to the asbestos ban.

In [33]:
#Find the homes with at least 2 bedrooms and at least one bathroom
df_clean[(df_clean['bedrooms'] >= 2) & (df_clean['bathrooms'] >= 1)]

id        date      price  bedrooms  bathrooms  sqft_living  \
0      7399300360   5/24/2022   675000.0         4        1.0         1180   
1      8910500230  12/13/2021   920000.0         5        2.5         2770   
2      1180000275   9/29/2021   311000.0         6        2.0         2880   
3      1604601802  12/14/2021   775000.0         3        3.0         2160   
4      8562780790   8/24/2021   592500.0         2        2.0         1120   
...           ...         ...        ...       ...        ...          ...   
30150  7834800180  11/30/2021  1555000.0         5        2.0         1910   
30151   194000695   6/16/2021  1313000.0         3        2.0         2020   
30152  7960100080   5/27/2022   800000.0         3        2.0         1620   
30153  2781280080   2/24/2022   775000.0         3        2.5         2570   
30154  9557800100   4/29/2022   500000.0         3        1.5         1200   

       sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
0          7140     1.0         NO        NO  ...         40     1969   
1          6703     1.0         NO        NO  ...        240     1950   
2          6156     1.0         NO        NO  ...          0     1956   
3          1400     2.0         NO        NO  ...        270     2010   
4           758     2.0         NO        NO  ...         30     2012   
...         ...     ...        ...       ...  ...        ...      ...   
30150      4000     1.5         NO        NO  ...        210     1921   
30151      5800     2.0         NO        NO  ...        520     2011   
30152      3600     1.0         NO        NO  ...        110     1995   
30153      2889     2.0         NO        NO  ...        100     2006   
30154     11058     1.0         NO        NO  ...          0     1965   

      yr_renovated                                            address  \
0                0  2102 Southeast 21st Court, Renton, Washington ...   
1                0  11231 Greenwood Avenue North, Seattle, Washing...   
2                0  8504 South 113th Street, Seattle, Washington 9...   
3                0  4079 Letitia Avenue South, Seattle, Washington...   
4                0  2193 Northwest Talus Drive, Issaquah, Washingt...   
...            ...                                                ...   
30150            0  4673 Eastern Avenue North, Seattle, Washington...   
30151            0  4131 44th Avenue Southwest, Seattle, Washingto...   
30152            0  910 Martin Luther King Jr Way, Seattle, Washin...   
30153            0  17127 114th Avenue Southeast, Renton, Washingt...   
30154            0  18615 7th Avenue South, Burien, Washington 981...   

             lat       long  postal_code  grade_code  condition_code  \
0      47.461975 -122.19052        98055           7               4   
1      47.711525 -122.35591        98133           7               3   
2      47.502045 -122.22520        98178           7               3   
3      47.566110 -122.29020        98118           9               3   
4      47.532470 -122.07188        98027           7               3   
...          ...        ...          ...         ...             ...   
30150  47.664740 -122.32940        98103           8               4   
30151  47.565610 -122.38851        98116           7               3   
30152  47.610395 -122.29585        98122           7               3   
30153  47.449490 -122.18908        98055           8               3   
30154  47.435840 -122.32634        98148           7               3   

       price_sqft  
0      572.033898  
1      332.129964  
2      107.986111  
3      358.796296  
4      529.017857  
...           ...  
30150  814.136126  
30151  650.000000  
30152  493.827160  
30153  301.556420  
30154  416.666667  

[26932 rows x 29 columns]

In [34]:
#Filter out homes with less than 2 bedrooms 
df_clean = df_clean[df_clean['bedrooms'] >= 2]

Filtered to home with at least two rooms to ensure that the house is suiteable for a single family home, one room for the parents and one room for the children.

In [35]:
#Filter out the homes with less than 1 bathroom
df_clean = df_clean[df_clean['bathrooms'] >=1]

Filtered to homes with atleast one full bath. A place to both shower and use the toilet is important! 

In [36]:
#Filter to homes with only 1 floor
df_clean = df_clean[df_clean['floors'] == 1]

Filtered to homes with only 1 floor to focus on the safety of children. Stairs can be scary!

In [37]:
#Filter out homes built after 1977
df_clean = df_clean[df_clean['yr_built'] > 1977]

The use of asbestos was banned in 1977. We have filtered the dataset to homes built after 1977 to ensure the house does not contain any asbestos.

In [38]:
#Filter to homes that meet building codes - Grade 6 or higher
df_clean = df_clean[df_clean['grade_code'] >= 6]

Building grade of 6 is the lowest grade that meets building codes. We want to make sure the house is of code and can surpass inspection.

In [39]:
#Filter to homes that are of average and above quality - Condition grade 3 or higher
df_clean = df_clean[df_clean['condition_code'] >= 3]

In [40]:
df_clean['condition_code'].unique()

array([4, 3, 5])

In [41]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2731 entries, 21 to 30152
Data columns (total 29 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              2731 non-null   int64  
 1   date            2731 non-null   object 
 2   price           2731 non-null   float64
 3   bedrooms        2731 non-null   int64  
 4   bathrooms       2731 non-null   float64
 5   sqft_living     2731 non-null   int64  
 6   sqft_lot        2731 non-null   int64  
 7   floors          2731 non-null   float64
 8   waterfront      2731 non-null   object 
 9   greenbelt       2731 non-null   object 
 10  nuisance        2731 non-null   object 
 11  view            2731 non-null   object 
 12  condition       2731 non-null   object 
 13  grade           2731 non-null   object 
 14  heat_source     2731 non-null   object 
 15  sewer_system    2731 non-null   object 
 16  sqft_above      2731 non-null   int64  
 17  sqft_basement   2731 non-null  

In [42]:
df_clean['price'].describe()

count    2.731000e+03
mean     8.426668e+05
std      3.869865e+05
min      4.118100e+04
25%      5.700000e+05
50%      7.300000e+05
75%      1.011000e+06
max      2.301000e+06
Name: price, dtype: float64

In [43]:
df_clean['yr_built'].describe()

count    2731.000000
mean     1987.685829
std         9.918578
min      1978.000000
25%      1980.000000
50%      1985.000000
75%      1992.000000
max      2022.000000
Name: yr_built, dtype: float64

In [44]:
df_clean.head()

id       date      price  bedrooms  bathrooms  sqft_living  \
21  2310000170  2/28/2022   750000.0         3        2.0         1590   
25  2095600170  10/8/2021   580000.0         3        3.0         2020   
26  8682282030   5/4/2022  2000000.0         3        3.0         2700   
34  5727500011  11/3/2021   785000.0         3        2.0         1350   
39  1421059003  8/25/2021  1680000.0         3        2.5         5200   

    sqft_lot  floors waterfront greenbelt  ... sqft_patio yr_built  \
21      7754     1.0         NO       YES  ...          0     1990   
25      4482     1.0         NO        NO  ...        120     1992   
26      7694     1.0         NO        NO  ...        640     2006   
34      7354     1.0         NO        NO  ...        120     1985   
39    206039     1.0         NO        NO  ...        670     2007   

   yr_renovated                                            address        lat  \
21            0  27307 226th Avenue Southeast, Maple Valley, Wa...  47.357065   
25            0  22609 102nd Place Southeast, Kent, Washington ...  47.399470   
26            0  23837 NE Adair Rd, Redmond, Washington 98053, ...  47.708760   
34            0  16710 Meridian Avenue North, Shoreline, Washin...  47.751020   
39            0  15916 Southeast 326th Street, Auburn, Washingt...  47.310595   

         long  postal_code  grade_code  condition_code  price_sqft  
21 -122.04067        98038           7               4  471.698113  
25 -122.20413        98031           7               3  287.128713  
26 -122.02073        98053           9               3  740.740741  
34 -122.33447        98133           7               5  581.481481  
39 -122.12938        98092          10               3  323.076923  

[5 rows x 29 columns]

### Export this cleaned dataset as a .csv, this will be our "master dataset"

In [45]:
df_clean.to_csv('data_cleaned.csv')

In [51]:
data_relevant = df_clean.copy()

In [52]:
data_relevant = data_relevant.drop(columns=['id', 'date', 'waterfront', 'greenbelt', 'nuisance', 'view',
                                  'condition', 'grade', 'heat_source', 'sewer_system','sqft_above','sqft_basement',
                                  'yr_renovated', 'address', 'lat', 'long'])

data_relevant.head()

price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
21   750000.0         3        2.0         1590      7754     1.0   
25   580000.0         3        3.0         2020      4482     1.0   
26  2000000.0         3        3.0         2700      7694     1.0   
34   785000.0         3        2.0         1350      7354     1.0   
39  1680000.0         3        2.5         5200    206039     1.0   

    sqft_garage  sqft_patio  yr_built  postal_code  grade_code  \
21          440           0      1990        98038           7   
25          600         120      1992        98031           7   
26          530         640      2006        98053           9   
34          460         120      1985        98133           7   
39         1080         670      2007        98092          10   

    condition_code  price_sqft  
21               4  471.698113  
25               3  287.128713  
26               3  740.740741  
34               5  581.481481  
39               3  323.076923

### Export this cleaned dataset with only relevant columns as a .csv

In [53]:
data_relevant.to_csv('data_relevant.csv')